### PR TITLE
Inconsistent SSA state after 'CallRewriter' running

### PR DESCRIPTION
**Pull request by ptomin**
- Move DefineUninitializedIdentifiers from IndirectCallRewriter to SsaMutator
- Validate SSA state after running of `CallRewriter` tests. It discovers inconsistent SSA state after `CallRewriter` running (two unit test failed). The fix is coming...

### CODE REVIEW

**Comment by ptomin** [comment]
I have created more unit tests to reproduce inconsistent SSA state after `CallRewriter` running. 
Now 4 unit tests failed
The fix is coming...

**Comment by uxmal** [comment]
Ok Good work.  I have a few endianness fixes (#727) that are in the pipeline. I will wait until you got this stable.

**Comment by ptomin** [comment]
e2da4a2 is final version. All tests passed. Regression test outputs were changed. Please review it.

**Comment by uxmal** [comment]
Will look in a couple hours

File: src/tests/Analysis/CoaDataConstraint.exp [hunk 1]
```diff
@@ -9,8 +9,8 @@ ds:ds
           Mem11[ds:0x0202:word16] = Mem9[ds:si + 0x0202:word16]
           Mem14[ds:0x0204:word16] = Mem11[ds:0x0204:word16] + Mem11[ds:si + 0x0204:word16]
           ax_16 = Mem14[ds:0x0200:word16]
-          fn0C00_002A(ds)
           Mem17[ds:0x0200:word16] = ax_16
+          fn0C00_002A(ds)
```
[the remark this answers is not in the document]
Comment (ptomin) [reply src/tests/Analysis/CoaDataConstraint.exp]: This is not regression. It's rearrangement of uses list.
Old version:
```
          Mem14[ds:0x0204:word16] = Mem11[ds:0x0204:word16] + Mem11[ds:si + 0x0204:word16]
          ax_16 = Mem14[ds:0x0200:word16]
          fn0C00_002A(ds)
          Mem17[ds:0x0200:word16] = ax_16
          Mem11[ds:0x0202:word16] = Mem9[ds:si + 0x0202:word16]
```
New version
```
          Mem14[ds:0x0204:word16] = Mem11[ds:0x0204:word16] + Mem11[ds:si + 0x0204:word16]
          ax_16 = Mem14[ds:0x0200:word16]
          Mem17[ds:0x0200:word16] = ax_16
          fn0C00_002A(ds)
          Mem11[ds:0x0202:word16] = Mem9[ds:si + 0x0202:word16]
```

File: src/Decompiler/Analysis/SsaMutator.cs [hunk 2]
```diff
@@ -120,5 +120,29 @@ public Statement InsertStatementAfter(Instruction instr, Statement stmAfter)
...
+            var trashedSids = call.Definitions.Select(d => (Identifier) d.Expression)
+                .Select(id => ssa.Identifiers[id])
+                .Where(sid => sid.DefStatement == null);
+            foreach (var sid in trashedSids)
+            {
+                DefineUninitializedIdentifier(stm, sid);
+            }
+        }
+
+        private void DefineUninitializedIdentifier(
```
[the remark this answers is not in the document]
Comment (ptomin) [reply src/Decompiler/Analysis/SsaMutator.cs]: Done in 5671338

**Comment by uxmal** [comment]
OK. my mistake. Thanks for keeping me honest.

**Comment by ptomin** [comment]
@uxmal Maybe uses should be sorted before printing to produce stable test output

**Comment by uxmal** [comment]
Since the `SsaState.Write` is never used during normal operation, it's not performance sensitive. I suppose the uses could be sorted by calling `ToString()` on each use and sorting on the resulting strings.

**Pull request merged**